### DISCUSSION TITLE
Vercel Postgres Python

### DISCUSSION THREAD

**Discussion by GitCat3** [vercel/community#2502] [Help]
### Summary

I am trying to connect to the postgres database using the python library psycopg2, however it for some reason does not want to connect using the url that is on the vercel page. Does anyone have a working instance of this that they would be willing to share? Thanks!

**Comment by adam-bartholomew** [comment]
I am using a vercel postgres database with my Flask application. I had a recurring issue with the connection URL and I needed to change how the URL started. It needed to start as "postgresql", rather than "postgre", which is what vercel's default environment variable had, in order for SQLAlchemy to work. I also believe Vercel complains if you have psycopg2 in your requirements.txt file, so you might want to try replacing it with the psycopg2-binary package. If you want to post your code/repository it will be easier to try and see what the issue is.

My Suggestions:
1 - Check that the environment variables you are using are correct for how you want to connect to the database.
2 - Try using the psycopg2-binary package in place of psycopg2.
3 - Make sure that you aren't using anything from a version of Python that is later than 3.9.

Here is my repository: https://github.com/adam-bartholomew/encode-message
Here is my live application: https://encode-message.vercel.app/

**Reply by bastianparedes** [reply]
This worked for my
